lib/mp4/transmuxer.js

```diff
@@ -1100,23 +1180,39 @@ Transmuxer = function(options) {
 
   // feed incoming data to the front of the parsing pipeline
   this.push = function(data) {
+    if (hasFlushed) {
+      var isAac = isLikelyAacData(data);
+
+      if (isAac && this.currentPipeline_ !== 'aac') {
+        if (this.currentPipeline_) {
+          this.teardownTsPipeline();
+        }
+        this.setupAacPipeline();
+      } else if (!isAac && this.currentPipeline_ !== 'ts') {
+        if (this.currentPipeline_) {
+          this.teardownAacPipeline();
+        }
+        this.setupTsPipeline();
+      }
+      hasFlushed = false;
+    }
     headOfPipeline.push(data);
   };
 
   // flush any buffered data
   this.flush = function() {
+    hasFlushed = true;
     // Start at the top of the pipeline and flush all pending work
     headOfPipeline.flush();
   };
 
-  // Re-emit any data coming from the coalesce stream to the outside world
-  coalesceStream.on('data', function (data) {
+  emitData = function (data) {
```

Comment:
I'm not really a fan of these helper functions for single method calls. Totally subjective, I know.